libs/react-components/src/tedi/components/buttons/button-content/button-content.tsx

```diff
@@ -138,7 +138,15 @@ const InternalButtonContent = forwardRef(
       const iconBEM = cn(styles['tedi-btn__icon'], styles[`tedi-btn__icon--${location}`], {
         [styles['tedi-btn__spinner']]: isLoading,
       });
-      const defaultIconProps: Partial<IconProps> = { size: 18, className: iconBEM };
+
+      const isAnchor = as === 'a' || !!rest.href;
```

Comment:
Not all links will have a href. For example React routers Link uses `to` prop. We should use something else to make the distinction. When using a button, that will visually have link styles the icon should be inline as well. So maybe use visualType === 'link' ?

Reply:
Yes that's a good idea, no clue why I overthought it